### PR TITLE
Feat/support debian12

### LINKED ISSUE #505: Spike: Debian 12 as Mgmt Server
As operator, I want to reduce my dependency on Ubuntu/Canonical and want to avoid relying on it for the k8s capi management host.

Investigate whether we can do the bootstrapping on Debian 12 (bookworm) instead.
Ideally also getting rid of Canonical snaps ...

### PR DESCRIPTION
**Pull request by garloff**
This fixes #505.

### CODE REVIEW

**Review by mxmxchere — APPROVED** [review]
deployed just fine, lgtm. I did not run the conformance-tests.

**Comment by jschoone** [comment]
Sign-off in https://github.com/SovereignCloudStack/k8s-cluster-api-provider/pull/509/commits/9f07cbf4e08f10b4b548083a16007d53eebbf1a1 is missing

File: doc/configuration.md [hunk 1]
```diff
@@ -19,7 +19,7 @@ Parameters controlling the Cluster-API management server (capi management server
 |--------------------------|-----------------|------------|-------------------|-------------------------------------------------------------------------------|
 | `prefix`                 |                 | SCS        | `capi`            | Prefix used for OpenStack resources for the capi mgmt node                    |
 | `kind_flavor`            |                 | SCS        | `SCS-2V-4`        | Flavor to be used for the k8s capi mgmt server                                |
-| `image`                  |                 | SCS        | `Ubuntu 22.04`    | Image to be deployed for the capi mgmt server                                 |
+| `image`                  |                 | SCS        | `Ubuntu 22.04`    | Image for the capi mgmt server. Use `Ubuntu 22.04` or `Debian 12`             |
```
Comment (chess-knight) [inline doc/configuration.md:22]: Do we support Ubuntu 20.04 and Debian 11? I can see `image = Ubuntu 20.04` in some environments

**Review by chess-knight — COMMENTED** [review]
LGTM, just a minor question

**Comment by garloff** [comment]
> Sign-off in [9f07cbf](https://github.com/SovereignCloudStack/k8s-cluster-api-provider/commit/9f07cbf4e08f10b4b548083a16007d53eebbf1a1) is missing

Sigh, my bad.
I would (as an exception) manually set "DCO to pass" -- and I don't keep the individual sign-offs in our squash merges anyway ...

File: doc/configuration.md (hunk 1 above)
Comment (chess-knight) [reply doc/configuration.md:22]: Thanks for the explanation, I am fine with that

**Pull request merged**